Source/Core/DolphinQt/Host.cpp

```diff
@@ -1,30 +1,19 @@
-// Copyright 2014 Dolphin Emulator Project
+// Copyright 2015 Dolphin Emulator Project
 // Licensed under GPLv2+
 // Refer to the license.txt file included.
 
-#include <cstring>
+#include <QDebug>
 
-#include <QApplication>
-#include <QResizeEvent>
-
-#include "Common/MsgHandler.h"
+#include "MainWindow.h"
 #include "Core/Host.h"
 
-#include "DolphinQt/MainWindow.h"
+void Host_Message(int id) {}
 
-void Host_Message(int id)
-{
-	// TODO
-}
-
-void Host_UpdateMainFrame()
-{
-	// TODO
-}
+void Host_UpdateMainFrame() {}
 
 void Host_UpdateTitle(const std::string& title)
 {
-	// TODO
+	g_main_window->setWindowTitle(QString::fromStdString(title));
```

Comment:
This is illegal, because `g_main_window` is on a different thread than the thread that calls the `Host_*` functions, so without locking this is illegal. And because Qt does all sorts of stuff internally that are unlockable, the only way to do this is with events. Which I already did in #3032.